src/metabase/models/setting.clj

```diff
@@ -617,10 +618,11 @@
   (let [{:keys [cache? getter enabled? default]} (resolve-setting setting-definition-or-name)
         disable-cache?                           (not cache?)]
     (if (or (nil? enabled?) (enabled?))
-      (if (= *disable-cache* disable-cache?)
-        (getter)
+      ;; optimization: only bind if the value will change. Saves FRACTIONS OF A SECOND
+      (if (and disable-cache? (not *disable-cache*))
         (binding [*disable-cache* disable-cache?]
-          (getter)))
+          (getter))
+        (getter))
       default)))
```

Comment:
There was actually a bug here, if `*disable-cache*` was truthy it would get bound to `false` if the Setting was a normal cached Setting, which was super confusing. Fixed